server/controllers/stock/lots.js

```diff
@@ -92,6 +94,85 @@ async function update(req, res, next) {
   }
 }
 
+/**
+ * GET /lot_dupes/:label?/:inventory_uuid?/:initial_quantity?/:entry_date?/:expiration_date?
+ *
+ * @description
+ * Returns all lots with the given label or matching field(s)
+ * inventory_uuid, initial_quantity, entry_date, expiration_date
+ *
+ * TODO: After getting this working purge unneeded params
+ *
+ */
+function dupes(req, res, next) {
+  const wheres = [];
+  if (req.query.label) {
```

Comment:
By the way, we have a fairly elegant, if undocumented, way of composing these filters using [`lib/filter.js`](https://github.com/IMA-WorldHealth/bhima/blob/master/server/lib/filter.js).

In essence you set up your base query and pass it an options object, then tell it which options are valid to use as WHERE query parts.  As a bonus, it does SQL injection escapes.  Here is how you would implement it here:
```js

const FilterParser = require('path/to/lib/filter');

const detailsQuery = `SELECT * FROM lots `;

// make sure all UUIDS are correctly parsed
const options = db.convert(req.query, ['inventory_uuid', 'lot_uuid', 'depot_uuid', /* etc */ ]);

// pass all the req.query options to the filterparser
const filters = new FilterParser(options, { tableAlias : 'lots' });

// tell it that "label" property should be a LIKE query (full-text index)
filters.fullText('label');

// tell it that inventory_uuid should be exactly equal
filters.equals('inventory_uuid');

// quantity should also be equal
filters.equals('initial_quantity');

// finally, let's get the statement out:
const statement = filters.applyQuery(detailsQuery);
const params = filters.parameters();

// pass to db.exec to execute them
return db.exec(statement, params);
```

Reply:
This will ensure that you are correctly escaping SQL injection attacks and is also a lot more flexible.  We use it everywhere - you can draw inspiration from [cash.js](https://github.com/IMA-WorldHealth/bhima/blob/master/server/controllers/finance/cash.js#L128) which has a pretty good set of filters.

Reply:
Done. Thanks for the pointers!